src/lib/react/ReactComponent.hx

```diff
@@ -11,30 +11,30 @@ typedef ReactComponentProps = {
 /**
 	https://facebook.github.io/react/docs/react-component.html
 **/
-typedef ReactComponent = ReactComponentOf<Dynamic, Dynamic, Dynamic>;
-typedef ReactComponentOfProps<TProps> = ReactComponentOf<TProps, Dynamic, Dynamic>;
-typedef ReactComponentOfState<TState> = ReactComponentOf<Dynamic, TState, Dynamic>;
-typedef ReactComponentOfRefs<TRefs> = ReactComponentOf<Dynamic, Dynamic, TRefs>;
-typedef ReactComponentOfStateAndRefs<TState, TRefs> = ReactComponentOf<Dynamic, TState, TRefs>;
-typedef ReactComponentOfPropsAndState<TProps, TState> = ReactComponentOf<TProps, TState, Dynamic>;
-typedef ReactComponentOfPropsAndRefs<TProps, TRefs> = ReactComponentOf<TProps, Dynamic, TRefs>;
+typedef ReactComponent = ReactComponentOf<Dynamic, Dynamic>;
+
+typedef ReactComponentOfProps<TProps> = ReactComponentOf<TProps, Void>;
+typedef ReactComponentOfState<TState> = ReactComponentOf<Void, TState>;
+
+#if react_comp_legacy
+// Keep the old ReactComponentOfPropsAndState typedef available for a few versions
+// But now we should use ReactComponentOf<TProps, TState> directly
+typedef ReactComponentOfPropsAndState<TProps, TState> = ReactComponentOf<TProps, TState>;
+#end
 
 #if (!react_global)
 @:jsRequire("react", "Component")
 #end
 @:native('React.Component')
 @:keepSub
 @:autoBuild(react.ReactComponentMacro.build())
-extern class ReactComponentOf<TProps, TState, TRefs>
+extern class ReactComponentOf<TProps, TState>
 {
 	var props(default, null):TProps;
 	var state(default, null):TState;
-	var context(default, null):Dynamic;
 
-	/**
-		https://facebook.github.io/react/docs/refs-and-the-dom.html
-	**/
-	var refs(default, null):TRefs;
+	@:deprecated
+	var context(default, null):Dynamic;
```

Comment:
Maybe instead of `@:deprecated` we keep them behind `-D react_comp_legacy`.

Also maybe `-D react_deprecated_api` is a better name :)

Reply:
Sure :)

As for the define name, I agree that it sucks. `react_deprecated_api` is not really future proof, though. Maybe two different defines, like `react_deprecated_refs` and `react_deprecated_context`?

Reply:
I went ahead and did it, feel free to request a define name change if that doesn't suit you.